crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -1304,6 +1305,51 @@ impl<'w, 's> SystemParamFetch<'w, 's> for SystemChangeTickState {
     }
 }
 
+#[derive(Debug)]
+pub struct SystemName {
+    name: Cow<'static, str>,
+}
+
+impl SystemName {
+    pub fn name(&self) -> &Cow<'static, str> {
+        &self.name
+    }
+}
+
+impl<'a> SystemParam for SystemName {
+    type Fetch = SystemNameState;
+}
+
+// SAFETY: Only reads World entities
+unsafe impl ReadOnlySystemParamFetch for SystemNameState {}
+
+/// The [`SystemParamState`] of [`Entities`].
+#[doc(hidden)]
+pub struct SystemNameState;
+
+// SAFETY: no component value access
+unsafe impl SystemParamState for SystemNameState {
+    fn init(_world: &mut World, _system_meta: &mut SystemMeta) -> Self {
```

Comment:
Can the system's name ever change?

If not, why don't we fetch it here, then use references all the way down? (If doing so, ignore my other comment about the owned->owned path)

(If so, can we remove that possibility?)

Reply:
I think that makes the most sense, we should be able to get a static str anyways so this could just store that.